src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class AuthenticationControllerIntegrationTest {
+    private static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private AuthenticationService authenticationService;
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+    }
+
+    @BeforeEach
+    void setUp(@Autowired DataSource dataSource) {
+        setupDatabase(dataSource);
+    }
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/controller"
+                            + "/authentication/remove-from-users_roles.sql")
+            );
+
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/controller"
+                            + "/authentication/remove-from-roles.sql")
+            );
+
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/controller"
+                            + "/authentication/remove-from-users.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void setupDatabase(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/controller"
+                            + "/authentication/add-user-to-users-table.sql")
+            );
+        }
+    }
+
+    @Test
+    @DisplayName("Register new user")
+    void register_validRequest_Success() throws Exception {
+        UserRegistrationRequest request = new UserRegistrationRequest()
+                .setEmail("newuser@test.com")
+                .setPassword("Newuser12345")
+                .setRepeatPassword("Newuser12345")
+                .setFirstName("User")
+                .setLastName("New");
+
+        UserResponseDto expected = new UserResponseDto()
+                .setId(3L)
+                .setEmail(request.getEmail())
+                .setFirstName(request.getFirstName())
+                .setLastName(request.getLastName());
+
```

Comment:
You can create template request, users, etc in some common file and use it every test what you need for increase readability